### ISSUE TITLE
Cross compile V7 and failed to run on mips-32bit

### ISSUE THREAD

**Issue by xxiao** [cesanta/v7#547]
./v7: v7.c: 9223: calc_llen: Assertion `n <= (int) sizeof(len)' failed.
Aborted

**Comment by goniz** [comment]
I'm experiencing this problem as well.
Can a fix be implemented easily?

**Comment by mmikulicic** [comment]
big endian?

**Comment by goniz** [comment]
Yes, big endian.

**Comment by goniz** [comment]
Is this a 32bit vs 64bit issue? or a big endian vs little endian issue?

**Comment by dimonomid** [comment]
v7 runs perfectly fine on both 64bit and 32bit. This is a big-endian issue.

**Comment by goniz** [comment]
running `./v7_mips -e 'print(1);'` after commit 91118131613f78037df1c71126e388d1a01ed443 give the following output:

"cannot find exception TypeError
Exception creation throws an exception TypeError: cannot read property 'prototype' of undefined
Assertion failed: v7_is_object(v) (v7.c: get_object_struct: 17840)"

**Comment by dimonomid** [comment]
Yes, we're aware of the issues on 64-bit platforms; honestly we never addressed them yet. But yes, this thing is important, and we plan to fix it in July 2016 (can't do that earlier due to our local deadlines). We'll report here once we have any progress on that.

**Comment by goniz** [comment]
Thank you for the quick response, July sounds great!
I will be following your progress eagerly :+1:

**Comment by goniz** [comment]
Looking forward for the fix :)
